src/Components/Web/src/Forms/InputDate.cs

```diff
@@ -70,6 +71,7 @@ protected override bool TryParseValueFromString(string? value, [MaybeNull] out T
 
             if (success)
             {
+                Debug.Assert(result != null);
```

Comment:
This is a bit quirky but isn't really a problem because it's only a `Debug.Assert`.

As per the logic above (lines 58-69), we know for sure that `TValue` is either `DateTime` or `DateTimeOffset`, neither of which are capable of holding the value `null`.

Reply:
I guess you're only doing this to satisfy the compiler's static analysis, in which case cool, so I'll just mark this thread as resolved :)